pynestml/codegeneration/nest_code_generator.py

```diff
@@ -908,8 +910,8 @@ def update_symbol_table(self, neuron) -> None:
         """
         SymbolTable.delete_model_scope(neuron.get_name())
         symbol_table_visitor = ASTSymbolTableVisitor()
-        symbol_table_visitor.after_ast_rewrite_ = True
         neuron.accept(symbol_table_visitor)
+        CoCosManager.check_cocos(neuron, after_ast_rewrite=True)
```

Comment:
Should the cococs be checked inside each code generator? Shouldn't it rather happen between transformation and code generation stages?

Reply:
Indeed, the primary coco checking call (to ``CoCosManager.check_cococs()``) is in the frontend. This is an additional call (with ``after_ast_rewrite=True``) after model transformations, as an extra sanity check.